src/ext/graphql/transmodelapi/schema.graphql

```diff
@@ -1753,3 +1755,23 @@ input TriangleFactors {
   "How important is time expressed as a fraction of 1. Note that what this really optimises for is distance (even if that means going over terrible surfaces, so I might be slower than the safe route)."
   time: Float!
 }
+
+"A collection of selectors for what lines/trips should be included in / excluded from search"
+input TripFilterInput {
+  "A list of selectors for what lines/trips should be excluded during the search. If line/trip matches with at least one selector it will be excluded."
+  not: [TripFilterSelectInput]
+  "A list of selectors for what lines/trips should be allowed during search. I order to be accepted a trip/line has to match with at least one selector.An empty list means that everything should be allowed. "
+  select: [TripFilterSelectInput]
+}
+
+"A list of selectors for filter allow-list / exclude-list. An empty list means that everything is allowed. A trip/line will match with selectors if it matches with all non-empty lists."
+input TripFilterSelectInput {
+  "Set of ids for authorities that should be included in/excluded from search"
+  authorities: [ID] = []
+  "Set of ids for lines that should be included in/excluded from search"
+  lines: [ID] = []
+  "Set of ids for service journeys that should be included in/excluded from search"
+  serviceJourneys: [ID] = []
+  "The allowed modes for the transit part of the trip. Use an empty list to disallow transit for this search. If the element is not present or null, it will default to all transport modes."
+  transportModes: [TransportModes]
+}
```

Comment:
All elements should be non-null here as well:
```
authorities: [ID!] = []
```

Reply:
done